src/sentry/models/artifactbundle.py

```diff
@@ -134,41 +133,20 @@ class Meta:
 
         index_together = (("project_id", "release_name", "dist_name"),)
 
-    def update_flat_file_index(self, file_contents: str):
-        from sentry.models import File
+    def _nodestore_id(self) -> str:
+        return f"bundle_index:{self.project_id}:{self.id}"
 
-        with atomic_transaction(
-            using=(router.db_for_write(File), router.db_for_write(ArtifactBundleFlatFileIndex))
-        ):
-            current_file = self.flat_file_index
+    def update_flat_file_index(self, data: str):
+        nodestore.set_bytes(self._nodestore_id(), data.encode())
```

Comment:
Do we know how node store applies `set` operations? We need to be careful about race conditions, especially since we are preventing them via locks which is quite fragile.

Reply:
As you mention we have an explicit lock around the read-modify-write operation.
The question rather is if we can trust that the *write* is committed and visible to readers when we release the lock.